src/gitlab/configuration.ts

```diff
@@ -0,0 +1,224 @@
+import { snake } from 'case';
+import { Component } from '../component';
+import { Project } from '../project';
+import { YamlFile } from '../yaml';
+import {
+  Default,
+  Include,
+  IncludeItem,
+  Job,
+  Service,
+  VariableConfig,
+  Workflow,
+} from './configuration-model';
+
+/**
+ * CI for GitLab.
+ * A CI is a configurable automated process made up of one or more stages/jobs.
+ * @see https://docs.gitlab.com/ee/ci/yaml/
+ */
+export class CiConfiguration extends Component {
+  public readonly project: Project;
+  public readonly name: string;
+  /** Path to CI file */
+  public readonly path: string;
+  /**
+       * The workflow YAML file.
+       */
+  public readonly file: YamlFile;
+  // /* Defines scripts that should run *after* the job. Can be set globally or per job. */
+  // private afterScript?: Array<string[] | string>;
+  // /* Defines scripts that should run *before* the job. Can be set globally or per job. */
+  // private beforeScript?: Array<string[] | string>;
+  // private cache?: Cache;
+  private default?: Default;
+  // private image?: Image | string;
+  /**
+       * Can be `IncludeItem` or `IncludeItem[]`. Each `IncludeItem` will be a string, or an
+       * object with properties for the method if including external YAML file. The external
+       * content will be fetched, included and evaluated along the `.gitlab-ci.yml`.
+       */
+  public readonly include: Array<IncludeItem | string> = [];
+  /**
+       * A special job used to upload static sites to Gitlab pages. Requires a `public/` directory
+       * with `artifacts.path` pointing to it.
+       */
+  private pages?: Job;
+  public readonly services: Array<Service | string> = [];
+  /**
+       * Groups jobs into stages. All jobs in one stage must complete before next stage is
+       * executed. Defaults to ['build', 'test', 'deploy'].
+       */
+  public readonly stages: string[] = [];
+  public readonly variables: Record<string, number | VariableConfig | string> =
+    {};
+  private workflow?: Workflow;
+  public readonly jobs: Record<string, Job> = {};
+
+  constructor(project: Project, name: string) {
```

Comment:
I think it should be possible to configure many of the options at construction time, no?
```suggestion
  constructor(project: Project, name: string, options: CiConfigurationOptions) {
```

Same goes for the GitlabConfiguration and NestedConfiguration classes.

Reply:
Resolved in https://github.com/projen/projen/pull/1357/commits/6d42ed0b3bd482a03fb6924aff5f964a01fd5f90